### PR TITLE
Terminology tooltip in tables: fix display issue

### PR DESCRIPTION
**Pull request by c0ffeeca7**
## Proposed change

- fix display issue in tables where 'learn more' link jumps on a new table line
image

- reported in #https://github.com/home-assistant/home-assistant.io/pull/34040#discussion_r1698393720

Solution:
image

## Type of change

- [x] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

## Summary by CodeRabbit

- **Bug Fixes**
	- Streamlined error handling to improve user feedback when arguments do not match expected syntax.
  
- **Improvements**
	- Enhanced tooltip rendering for consistent output and clarity.
	- Adjusted string handling to remove unnecessary whitespace and ensure proper link inclusion.

### CODE REVIEW

**Review by CFenner — APPROVED** [review]
👏 thanks for the quick fix!

File: plugins/terminology_tooltip.rb [hunk 1]
```diff
@@ -1,51 +1,51 @@
...
+        tooltip = "<span class='terminology-tooltip'>#{definition.strip}</span>"
 
         "<span class='terminology'>#{@text || @term}#{tooltip}</span>"
       end
 
       private
 
-      SYNTAX = %r!^(\w+?|".+?")(?:\s+(\w+|".+"))?$!.freeze
-
+      SYNTAX = /^(\w+?|".+?")(?:\s+(\w+|".+"))?$/
```
Comment (c0ffeeca7) [inline plugins/terminology_tooltip.rb:48]:
```suggestion
      SYNTAX = %r!^(\w+?|".+?")(?:\s+(\w+|".+"))?$!.freeze
```

**Review by frenck — APPROVED** [review]

File: plugins/terminology_tooltip.rb [hunk 2]
```diff
@@ -1,51 +1,51 @@
...
 
       def render(context)
-        @term.gsub!(/\"/, "")
-        entries = context.registers[:site].data["glossary"].select do |entry|
-          entry.key?("term") and (@term.casecmp(entry["term"]).zero? or (entry.key?("aliases") and entry["aliases"].any?{ |s| s.casecmp(@term)==0 }))
+        @term.gsub!(/"/, '')
+        entries = context.registers[:site].data['glossary'].select do |entry|
+          entry.key?('term') and (@term.casecmp(entry['term']).zero? or (entry.key?('aliases') and entry['aliases'].any? do |s|
+                                                                           s.casecmp(@term) == 0
+                                                                         end))
```
Comment (c0ffeeca7) [inline plugins/terminology_tooltip.rb:24]:
```suggestion
        @term.gsub!(/\"/, "")
        entries = context.registers[:site].data["glossary"].select do |entry|
          entry.key?("term") and (@term.casecmp(entry["term"]).zero? or (entry.key?("aliases") and entry["aliases"].any?{ |s| s.casecmp(@term)==0 }))
```

File: plugins/terminology_tooltip.rb [hunk 3]
```diff
@@ -1,51 +1,51 @@
...
+          Syntax error in tag 'term' while parsing the following options:
+
+          #{args}
+
+          Valid syntax:
+            {% term <term> [<text>] %}
+        MSG
+
+        @term = Regexp.last_match(1)
+        @text = Regexp.last_match(2)
```
Comment (c0ffeeca7) [inline plugins/terminology_tooltip.rb:16]:
```suggestion
        if args.strip =~ SYNTAX
          @term = Regexp.last_match(1)
          @text = Regexp.last_match(2)
        else
          raise SyntaxError, <<~MSG
            Syntax error in tag 'term' while parsing the following options:
            #{args}
            Valid syntax:
              {% term <term> [<text>] %}
          MSG
        end
```

**Pull request merged**